src/typeahead/typeahead.directive.ts

```diff
@@ -58,6 +58,8 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
    * true make sense if options array is large.
    */
   @Input() typeaheadAsync: boolean = void 0;
+  /** If enabled, options will not be filtered by field value on focus when typeaheadMinLength is 0 */
+  @Input() typeaheadShowAllOnFocus = false;
```

Comment:
I think, it will be not so ofter used property, so, it's more suitable to add it via typeaheadConfig